### PR TITLE
Migrate Bulk POST API's and Files API to platform

### PR DESCRIPTION
**Pull request by Sravanksk**

### CODE REVIEW

File: apps/mappings/tasks.py [hunk 1]
```diff
@@ -49,8 +50,10 @@ def create_fyle_categories_payload(categories: List[DestinationAttribute], works
         if category.value not in existing_category_names:
             payload.append({
                 'name': category.value,
+                "sub_category": " ",
```
Comment (NileshPant1999) [inline apps/mappings/tasks.py:53]: single quotes to maintain consistency

File: apps/xero/tasks.py [hunk 2]
```diff
@@ -60,10 +67,29 @@ def load_attachments(xero_connection: XeroConnector, ref_id: str, ref_type: str,
     """
     try:
         fyle_credentials = FyleCredential.objects.get(workspace_id=expense_group.workspace_id)
-        expense_ids = expense_group.expenses.values_list('expense_id', flat=True)
-        fyle_connector = FyleConnector(fyle_credentials.refresh_token)
-        attachments = fyle_connector.get_attachments(expense_ids)
```
Comment (NileshPant1999) [inline apps/xero/tasks.py:65]: @Sravanksk we can also remove get_attachments function from connector file ??

**Review by NileshPant1999 — APPROVED** [review]

File: apps/xero/tasks.py [hunk 3]
```diff
@@ -60,10 +66,29 @@ def load_attachments(xero_connection: XeroConnector, ref_id: str, ref_type: str,
...
+        if len(files_list):
+            payload = {
+                "data": files_list
+            }
+
+            attachments = platform.connection.v1beta.admin.files.bulk_generate_file_urls(payload=payload)['data']
+
+            if attachments:
+                for attachment in attachments:
+                    attachment['download_url'] = get_as_base64(attachment['download_url'])
```
Comment (ashwin1111) [inline apps/xero/tasks.py:88]: can we create a helper function in platform connector for all these operations?

File: apps/xero/tasks.py [hunk 4]
```diff
@@ -795,7 +819,15 @@ def process_reimbursements(workspace_id):
...
-        fyle_connector.post_reimbursement(reimbursement_ids)
+        reimbursements_list = []
+        for reimbursement_id in reimbursement_ids:
+            reimbursement_object = {'id': reimbursement_id}
+            reimbursements_list.append(reimbursement_object)
+
+        payload = {
+            "data": reimbursements_list
+        }
+        platform.connection.v1beta.admin.reimbursements.bulk_post_reimbursements(payload)
```
Comment (ashwin1111) [inline apps/xero/tasks.py:830]: can we create a helper function in platform connector for all these operations?

**Review by ashwin1111 — APPROVED** [review]

File: apps/xero/tasks.py [hunk 5]
```diff
@@ -50,6 +51,11 @@ def get_or_create_credit_card_contact(workspace_id: int, merchant: str):
 
     return contact
 
+
+def get_as_base64(url):
```
Comment (NileshPant1999) [inline apps/xero/tasks.py:55]: @sravan not using this anywhere now, can remove

File: apps/xero/tasks.py [hunk 6]
```diff
@@ -4,6 +4,8 @@
 from datetime import datetime, timedelta
 from time import sleep
 from typing import List
+import base64
```
Comment (NileshPant1999) [inline apps/xero/tasks.py:7]: these too @Sravanksk

File: apps/xero/tasks.py [hunk 7]
```diff
@@ -4,6 +4,8 @@
...
+import requests
```
Comment (NileshPant1999) [inline apps/xero/tasks.py:8]: this too @Sravanksk

**Review by NileshPant1999 — APPROVED** [review]

**Pull request merged**